decisions/DR009-Locatable-content-traits.md

```diff
@@ -0,0 +1,207 @@
+context object is the mechanism that classifies calling context, and
+could contain information about supported formats/strategies for read
+-to allow the manager to customize its response. A consideration of the
+calling context is the standard approach for varying API responses based
+on the process environment.
+
+A manager may take care of re-locating data, and may be able to make the
+same entity data available via multiple means (eg: file on disk, http
+stream etc.).
+
+Traits are by design, atomic, in that adding a trait to an entity should
+not affect the data represented by an existing trait.
+
+## Options considered
+
+### Option 1
+
+An additional property-less `file` trait that when composed with a
+`locatableContent` trait holding an URL specialises the specification
+such that only URLs with the `file` scheme should be returned.
+
+#### Pros
+
+-   Can easily be appended to an existing trait set, removing the
+    requirement to have duplicate specifications for path or URL access
+    to the same entity data.
+-   Explicitly determines when a host requires data to be available
+    through the file system.
+-   More explicit representation of the filtering required during
+    browsing/etc.
+-   As the trait is additive, it can simply be omitted when URLs are
+    supported with no need to re-write data.
+
+#### Cons
+
+-   Introduces side-effects in trait composition, violating the design
+    principal that traits should be atomic.
+-   String manipulation overhead may be introduced to remove `file://`
+    prefix in path-based hosts.
+
+### Option 2
+
+A distinct `fileLocatableContent` trait that holds a `path`, used in
+place of the URL-holding `locatableContent` when filesystem-only access
+is required.
+
+#### Pros
+
+-   Explicitly determines when a host requires data to be available
+    through the file system.
+-   More explicit representation of the filtering required during
+    browsing/etc.
+-   No sting manipulation overhead in path-based hosts.
+
+#### Cons
+
+-   Requires duplicate specifications to be defined for the same entity
+    data to be accessed via path or URL, removing much of the benefits
+    gained from traits vs hierarchical specifications.
+-   Requires a manager to potentially re-write data depending on the
+    exact traits used for registration and resolution, converting
+    between paths/URLs.
+
+### Option 3
+
+A single `locatableContent` trait that holds a `location` property,
+whose content is determined by the options set in a
+`locatableContentReader` trait in the call's Context's locale.
+
+#### Pros
+
+-   Allows the host to declare the locator strategies it supports to
+    access content (eg: `path`, `URL`, etc.), which can influence
+    `resolve` and browsing.
+-   No sting manipulation overhead in path-based hosts.
+-   No need for duplicate specifications.
+-   Uses the API mechanism for adapting responses at runtime (the
+    calling context's locale).
+
+#### Cons
```

Comment:
[consider][minor] Precludes requesting both URL and file path in a single call (can't really think why you would want that though)

Reply:
Added [here](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/356/commits/e4bac68cafdfa0114e57aa1966972b132d1851f6#diff-f967dcd0760a5812bf92149546082da9b3010893cf1e97929bb3a3d03ab62714R141).